### DISCUSSION TITLE
[Question] IPF miller index tolerance degree

### DISCUSSION THREAD

**Discussion by labnewman** [mtex-toolbox/mtex#1401] [Ask Anything]
Hi community,
Sorry for new blood in coding Mtex.

I notice that when I bought single crystal copper from vendors, they will have a statement like this '(1,0,1 )+/- 2 degree'.
So it means that the single crystal is not exactly 100% (1,0,1).

As for my question, when I analyze data by MTEX. Usually, the miller index of IPF mapping will be very sensitive, which is no needed for me. 
Like the picture below, (0,1,12) is no needed for me. Can we change some tolerance degree and let the **IPF mapping color and miller index** (0,1,12) to be (0,0,1)? 
Which means let the software not so sensitive and more close to the three corner, (0,0,1), (0,1,1) and (-1,1,1)

![螢幕擷取畫面 2022-04-14 104923](https://user-images.githubusercontent.com/100294586/163304009-d933b8b5-370f-4423-9314-ceb5f13ac3f7.png)

**Comment by kilir** [comment]
Hi,
I think `Miller/gridify()` could do what you are looking for. Also, `Miller/round()` might be an option.
Cheers,
Rüdiger

**Reply by labnewman** [reply]
Hi kilir,
It's quite hard for me. Could you explain in detail or give me a example that i can change parameters?
I try to search gridify and round on MTEX website, but I don't know how to use the syntax.

https://mtex-toolbox.github.io/Miller.gridify.html
https://mtex-toolbox.github.io/orientation.round2Miller.html

Thanks!

**Reply by kilir** [reply]
Hi,
here are examples for Miller/gridify and Miller/round:
```
cs = crystalSymmetry('m-3m');
h = Miller.rand(1000,cs);
h = h.project2FundamentalRegion;
% option 1: approximation on an equispaced grid
hg = h.gridify('resolution', 5*degree)
plot(hg,'fundamentalRegion')
hold on
plot(h,'fundamentalRegion')
hold off

% option 2: rounding Millers (not equispaced!)
hr = h.round('maxHKL',5)
plot(hr,'fundamentalRegion')
hold on
plot(h,'fundamentalRegion')
hold off
```

Cheers,
Rüdiger

**Reply by labnewman** [reply]
Hi kilir,

I tried second one, and it comes out a picture like this.
![螢幕擷取畫面 2022-04-28 163750](https://user-images.githubusercontent.com/100294586/165713003-0164c0d2-89c1-498a-8f19-59e6aa4f688d.png)

But my RED point (0,1,12) disappeard. I want it become (001) and then show it on the IPF mapping.
Do you have a method to achieve that?

Thank you

newman

**Reply by kilir** [reply]
Yes, exactly.
Rounding hkl will not give an evenly spaced grid. Rounding (0,1,12) up to a maximum hkl of 6 gives (0,0,1) to me.
However, could it be taht you actually would rather rotate your sample till your [001] is parallel to z?
Cheers,
Rüdiger
